src/controllers/expenses.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const expensesService = require('../services/expenses');
+const usersService = require('../services/users');
+const { resetExpenses } = expensesService;
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+  let expenses = expensesService.getAll();
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === Number(userId));
+  }
+
+  if (categories) {
+    expenses = expenses
+      .filter(expense => categories.includes(expense.category));
+  }
+
+  if (from) {
+    expenses = expenses.filter(expense => (
+      new Date(expense.spentAt).getTime() > new Date(from).getTime()
+    ));
+  }
+
+  if (to) {
+    expenses = expenses.filter(expense => (
+      new Date(expense.spentAt).getTime() < new Date(to).getTime()
+    ));
+  }
```

Comment:
As for me - more logical to move filtration to service and only process request here